src/GafferScene/Preview/RendererAlgo.cpp

```diff
@@ -512,6 +513,21 @@ struct LocationOutput
 				updatedAttributes->members()[g_setsAttributeName] = boost::const_pointer_cast<InternedStringVectorData>( setsAttribute );
 			}
 
+			// TODO: clean up
+			CompoundObject::ObjectMap::const_iterator it = updatedAttributes->members().find( "linkedLightsSet" ); // TODO: naming needs to be determined
+			if( it != updatedAttributes->members().end() )
+			{
+				IECore::StringVectorDataPtr lightNames = new IECore::StringVectorData();
+				std::vector<std::string> &lightNamesWritable = lightNames->writable();
+
+				if( StringDataPtr lightsExpressionData = IECore::runTimeCast<StringData>( it->second ) )
+				{
+					GafferScene::ConstPathMatcherDataPtr linkedlightsSet = SetExpression::evaluateSetExpression( lightsExpressionData->readable(), scene );
```

Comment:
I have added a commit that takes a first stab at the caching. I'm not sure if that's what you had in mind and I'm particularly unsure of how much you dislike my use of `mutable` in this case.